Misreconstructed events when loading DL2 MC files

Yesterday, when doing some tests by not applying any intensity cut during the stereo stage, I got the following error:

```python
magicctapipe/io/tests/test_io_monly.py:200: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
magicctapipe/io/io.py:951: in load_mc_dl2_data_file
    event_table["reco_alt"] *= u.deg
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MaskedColumn name='reco_alt' dtype='float64' length=22>
74.05783472238909
74.28497526497506
               --
75.0970...11565059572
 72.5046340957998
74.16839589324137
73.90888016013854
 74.1882606020732
73.45272931733223
74.37786398967206
other = Unit("deg")

    def __imul__(self, other):
        """
        Multiply self by other in-place.
    
        """
        m = getmask(other)
        if self._mask is nomask:
            if m is not nomask and m.any():
                self._mask = make_mask_none(self.shape, self.dtype)
                self._mask += m
        elif m is not nomask:
            self._mask += m
        other_data = getdata(other)
        other_data = np.where(self._mask, other_data.dtype.type(1), other_data)
>       self._data.__imul__(other_data)
E       numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'multiply' output from dtype('O') to dtype('float64') with casting rule 'same_kind'

../miniconda3/envs/mcp-dev/lib/python3.11/site-packages/numpy/ma/core.py:4360: UFuncTypeError

```

this is the same error which was solved for real data some time ago by dropping the events with NaNs.

--- a/magicctapipe/io/io.py
+++ b/magicctapipe/io/io.py
@@ -940,6 +940,14 @@ def load_mc_dl2_data_file(input_file, quality_cuts, event_type, weight_type_dl2)
     # Get the mean DL2 parameters
     df_dl2_mean = get_dl2_mean(df_events, weight_type_dl2)
     df_dl2_mean.reset_index(inplace=True)
+    event_before_nan_filter = df_dl2_mean.shape[0]
+    df_dl2_mean.dropna(inplace=True)
+    event_after_nan_filter = df_dl2_mean.shape[0]
+
+    logger.info(
+        f"{event_before_nan_filter - event_after_nan_filter} "
+        f"stereo events removed because of NaN values."
+    )
 
     # Convert the pandas data frame to the astropy QTable
     event_table = QTable.from_pandas(df_dl2_mean)

